### DISCUSSION TITLE
Dot repeat seems to be broken

### DISCUSSION THREAD

**Discussion by darvelo** [mickael-menu/ShadowVim#55] [Help] [answered]
### Describe the issue

First of all thank you so much for making this project! It's so essential to have real vim in an Xcode workflow. 👏🏽 

Using dot repeat seems to only capture a part of the action. In the video below you can see I `ciw` to replace the word, then go a couple lines below and press `.` to repeat, but instead it deletes the word.

https://github.com/mickael-menu/ShadowVim/assets/523169/27fcb4cf-7d45-41d5-90f0-49b225204624

### How to reproduce?

1. Open a source file
2. `ciw` to replace a word
3. Use `.` to repeat the action on some other word

**Comment by mickael-menu** [comment, marked as answer]
Unfortunately, it's a side effect of having the Insert mode handled by Xcode instead of Neovim (for performance reasons, and to improve the auto-completion, auto-indentation, etc.).

A workaround is to explicitly set the input to Neovim when in Insert mode. You can do that with `:SVSetInputNvim`.

Personally, I have a binding on <kbd>C-n</kbd> that I use after doing `ciw`.

```viml
imap <C-n> <Cmd>SVSetInputNvim<CR>
```

Alternatively, you can override `ciw` to toggle the Neovim input on-the-fly. You will need to do that for all the mappings you use.

```viml
noremap ciw <Cmd>SVSetInputNvim<CR>ciw
```

Note that by using `SVSetInputNvim` in Insert mode, the Xcode auto-completion won't be optimal.

**Reply by mickael-menu** [reply]
More info about this issue: https://github.com/mickael-menu/ShadowVim/blob/6b6e5de64573a3c1f4d24d17b78cfff1899c0c0f/Documentation/Research/Mediator.md#insert-mode

**Comment by darvelo** [comment]
Thank you, this was very helpful. I started adding overrides for the movements like `ciw` and they're working great.
